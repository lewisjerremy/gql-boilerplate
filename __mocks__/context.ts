import { PrismaClient } from '@prisma/client'
import { mockDeep, DeepMockProxy } from 'jest-mock-extended'

export { Context } from '../src/context'

export type MockContext = {
  prisma: DeepMockProxy<PrismaClient>
}

export const createMockContext = (): MockContext => {
  return {
    prisma: mockDeep<PrismaClient>(),
  }
}
