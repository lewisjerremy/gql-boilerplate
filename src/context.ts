import { prisma } from './config/prisma'
import { PrismaClient } from '@prisma/client'

type Request = Record<string, unknown>

export interface Context {
  prisma: PrismaClient
  req: Request
}

export function createContext(req: Request) {
  return {
    ...req,
    prisma,
  }
}
