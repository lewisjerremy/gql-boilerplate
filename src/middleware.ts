import { schema } from './schema'
import { permissions } from './config/permissions'
import { applyMiddleware } from 'graphql-middleware'

const afterSchema = applyMiddleware(schema, permissions)

export { afterSchema }
