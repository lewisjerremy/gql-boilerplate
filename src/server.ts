import { ApolloServer } from 'apollo-server'
import { afterSchema } from './middleware'
import { createContext } from './context'

export const server = new ApolloServer({
  schema: afterSchema,
  context: createContext,
})
