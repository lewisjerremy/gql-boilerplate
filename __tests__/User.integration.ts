import { userData } from '../src/constants/user.data'
import { createContext } from '../src/context'

const ctx = createContext({})

async function reset() {
  await ctx.prisma.user.deleteMany()

  for (const user of userData) {
    await ctx.prisma.user.create({ data: user })
  }
}

beforeAll(async () => {
  await reset()
})

afterAll(async () => {
  await reset()
})

describe('User', () => {
  it('should persist data', async () => {
    const users = await ctx.prisma.user.findMany()
    expect(users).toMatchSnapshot()
  })
})
