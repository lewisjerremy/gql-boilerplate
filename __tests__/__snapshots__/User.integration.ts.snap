// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should persist data 1`] = `
Array [
  Object {
    "createdAt": 2022-02-01T07:00:00.000Z,
    "id": 1,
    "name": "Tim",
    "updatedAt": 2022-02-01T07:00:00.000Z,
  },
  Object {
    "createdAt": 2022-02-01T07:00:00.000Z,
    "id": 2,
    "name": "Robert",
    "updatedAt": 2022-02-01T07:00:00.000Z,
  },
  Object {
    "createdAt": 2022-02-01T07:00:00.000Z,
    "id": 3,
    "name": "John",
    "updatedAt": 2022-02-01T07:00:00.000Z,
  },
  Object {
    "createdAt": 2022-02-01T07:00:00.000Z,
    "id": 4,
    "name": "Mark",
    "updatedAt": 2022-02-01T07:00:00.000Z,
  },
  Object {
    "createdAt": 2022-02-01T07:00:00.000Z,
    "id": 5,
    "name": "Cam",
    "updatedAt": 2022-02-01T07:00:00.000Z,
  },
]
`;
